torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -344,6 +349,21 @@ class TensorPipeAgent : public RpcAgent {
   // Mutex to guard networkData_
   std::mutex networkDataMutex_;
 
+  struct CallCounter final {
+    CallCounter(TensorPipeAgent& agent, int32_t& counter)
+        : agent_(agent), counter_(counter) {
+      agent_.increaseCallCount(counter_);
+    }
+
+    ~CallCounter() {
+      agent_.decreaseCallCount(counter_);
+    }
```

Comment:
When defining a custom constructor and destructor, the compiler will still add an implicitly-declared copy constructor. Right now this doesn't cause problems, because we allocate the CallCounter in a shared_ptr and never copy it. However, keeping that copy constructor around introduces the risk that at some point someone will implicitly call it by accident and that will lead to hard-to-debug issues: the copy will not increase the count (because it will use a different constructor) however it will still decrease it (as it uses the same destructor). I'd suggest we explicitly delete the copy constructor and assignment operator, to be safe.